pallets/claim/src/lib.rs

```diff
@@ -0,0 +1,222 @@
+        /// Event emitted when a transaction has been stored.
+        ERC20TransactionStored(AccountId, EthereumTxHash, EthereumAddress, u64),
+        /// Event emitted when a transaction has been claimed.
+        ERC20TokenClaimed(AccountId, EthereumTxHash),
+        // ClaimDebug(AccountId, Vec<u8>, EthereumTxHash, EthereumAddress, EthereumAddress),
+    }
+);
+
+// Errors inform users that something went wrong.
+decl_error! {
+    pub enum Error for Module<T: Trait> {
+        /// Not administrator
+        NotAdministrator,
+        /// The transaction signature is invalid.
+        InvalidSignature,
+        /// The signer is not transaction sender.
+        NotTransactionSender,
+        /// The transaction hash doesn't exist
+        TxHashNotFound,
+        /// The transaction hash already exit
+        TxHashAlreadyExist,
+        /// The transaction has been claimed
+        TxAlreadyClaimed,
+        /// The transaction height less than end height
+        LessThanEndHeight,
+    }
+}
+
+// Dispatchable functions allows users to interact with the pallet and invoke state changes.
+// These functions materialize as "extrinsics", which are often compared to transactions.
+// Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+      	// Errors must be initialized if they are used by the pallet.
+		type Error = Error<T>;
+      	fn deposit_event() = default;
+
+      	#[weight = 0]
+      	pub fn set_admin(origin, admin: T::AccountId) -> dispatch::DispatchResult {
+      		ensure_root(origin)?;
+      		Admin_Id::<T>::put(admin);
+            Ok(())
+        }
+
+
+        #[weight = 0]
+        pub fn store_erc20_burned_transaction(origin, height: u64, eth_tx_hash: EthereumTxHash, eth_address: EthereumAddress, amount:u64) -> dispatch::DispatchResult {
+            let who = ensure_signed(origin)?;
+            // let admin = Admin_Id::<T>::get();
+            // ensure!(who == admin, Error::<T>::NotAdministrator)
+            ensure!(!BurnedTransactions::contains_key(&eth_tx_hash), Error::<T>::TxHashAlreadyExist);
+            let end_height = EndHeight::get();
+            if height > end_height {
+                EndHeight::put(height);
+            }
+            BurnedTransactions::insert(&eth_tx_hash, (eth_address.clone(), amount.clone()));
+            ClaimState::insert(&eth_tx_hash, false);
+            Self::deposit_event(RawEvent::ERC20TransactionStored(who, eth_tx_hash, eth_address, amount));
+            Ok(())
+        }
+
+
+        #[weight = 0]
+        pub fn claim_erc20_token(origin, eth_tx_hash: EthereumTxHash, eth_signature: EcdsaSignature) -> dispatch::DispatchResult {
+            ensure!(BurnedTransactions::contains_key(&eth_tx_hash), Error::<T>::TxHashNotFound);
+            ensure!(!ClaimState::get(&eth_tx_hash), Error::<T>::TxAlreadyClaimed);
+            let who = ensure_signed(origin)?;
```

Comment:
Move `ensure_signed` to the first line of code, because we don't want to run any unnecessary check before we make sure the transaction is signed. It may be unnecessary, but this is always a good practice.